streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -134,13 +136,21 @@ public void register(final StateStore store,
             restoreCallbacks.put(topic, stateRestoreCallback);
         } else {
             log.trace("Restoring state store {} from changelog topic {}", storeName, topic);
+
+            final StateStore stateStore =
+                store instanceof WrappedStateStore ? ((WrappedStateStore) store).inner() : store;
```

Comment:
We should update the tests for setting the `stateStore` and `recordConverter`

Reply:
I was able to add a test for `GlobalStateManagerImpl` -- also discovered a gap while trying to add a test for `ProcessorStateManagerImpl`: updating standby tasks was not covered by the converter -- added a test for this case.

However, I am not sure how I can test the converter for `ProcessorStateManagerImpl` restore case, because the `ProcessorStateManagerImpl` does not use the converter, but by the `TaskManager` -- however, it would be weird to test test in the `TaskManagerTest` because it has nothing to do with the task manager.

Thoughts?

I'll add some upgrade integration tests later, that will cover this case. Not sure if this would be sufficient?

Reply:
\cc @guozhangwang